src/utils/setUserScopePermission.js

```diff
@@ -0,0 +1,66 @@
+const { hasWrite } = require('./permissions');
+const { isPaginated } = require('./pagination');
+
+const KEY_NAME = 'scopePermission'; // is forced by this utils
+
+/**
+ * For every request it should return also the user scope, forced over this util.
+ * @param {ResultObject} result The request result, that should modified.
+ * @param {String ||  Array::Permission} permissions
+ * @param {String} permissionsAsString It is set directly for example in create request 'write'.
+ * @param {Array::Permission} permissionsAsArrayOfObjects To check the user scoped permissions and set it.
+ * @param {User} user The user object to set the permission scope.
+ * @returns {ResultObject}
+ */
+const setUserScopePermission = (result, permissions, user) => {
+	if (Array.isArray(permissions) && user) {
+		let perm = 'read';
+		if (hasWrite(permissions, user)) {
+			perm = 'write';
+		}
+		result[KEY_NAME] = perm;
+	} else if (typeof permissions === 'string') {
+		result[KEY_NAME] = permissions;
+	} else {
+		throw new Error('Wrong permissions or user add to setScopePermission', { permissions, user });
+	}
+	return result;
+};
+
+/**
+ * For every request it should return also the user scope, forced over this util.
+ * This is only for paginated find results, please use setUserScopePermission for other requests.
+ * !!! Is expensiv operation at the moment. !!!
+ * @param {PaginatedResultObject||Array::ResultObject} paginatedResult The request result, that should modified.
+ * @param {Array::Permission} permissions
+ * @param {User} user The user object to set the permission scope.
+ * @returns {PaginatedResultObject||Array::ResultObject}
+ */
+const setUserScopePermissionForFindRequests = (result, user, key = 'permissions') => {
```

Comment:
Wouldn't it easier to give the data object to the function above? When you call the function in the service you already know if data are paginated or not, so you do not have check it

Reply:
"function above" you mean setUserScopePermission then not. If you mean that the services must now that he can only pass paginated objects, ok. But the check if the result is paginated is adding in any situation. But at the moment it can handle array and paginated. In the other situation i throw a error if it is not paginated.